### PR TITLE
add istiod port into gateway

### PR DESCRIPTION
**Pull request by irisdingbj**

### CODE REVIEW

**Review by sdake — APPROVED** [review]
required for 2 of our 3 multicluster code scenarios. 

I am not certain if `15012` is protected by mutual TLS passthrough. Will approve for master, we can make a call on a 1.5 backport depending on the security implications of opening 15012 via the gateway.

@iris is 15012 protected by mutual TLS, or not protected?

**Review by sdake — APPROVED** [review]
approve

**Comment by irisdingbj** [comment]
> required for 2 of our 3 multicluster code scenarios.
> 
> I am not certain if `15012` is protected by mutual TLS passthrough. Will approve for master, we can make a call on a 1.5 backport depending on the security implications of opening 15012 via the gateway.
> 
> @Iris is 15012 protected by mutual TLS, or not protected?

It is mutual TLS

**Review by howardjohn — CHANGES_REQUESTED** [review]
I am worried this may open up some vulnerabilities. Are we exposing the debugging endpoint?

**Comment by sdake** [comment]
> this opens up huge security vulnerabilities

I was concerned about that problem in my review. Can you expand on the attack vectors? Istio is not well served by security through obscurity,. Understanding the attack vectors drives the development of a secure solution.

**Review by sdake — COMMENTED** [review]
@irisdingbj can you add the issue tracker(s) this fixes in commit message and github description field.

**Comment by howardjohn** [comment]
> I was concerned about that problem in my review. Can you expand on the attack vectors? Istio is not well served by security through obscurity,. Understanding the attack vectors drives the development of a secure solution.

Pilot is not designed to be secure to the open public. There are debug interfaces, no rate limits, etc that can easily be leveraged in an attack

**Comment by sdake** [comment]
Thanks for the answer. We can end the thread here.

File: manifests/gateways/istio-ingress/values.yaml [hunk 1]
```diff
@@ -92,6 +92,9 @@ gateways:
     - port: 15011
```
Comment (costinm) [inline manifests/gateways/istio-ingress/values.yaml]: Please also remove 15011 and 8060 - no longer needed. ( there is a quota on firewall rules and/or ports on some clouds)

**Review by costinm — APPROVED** [review]
May need to be backported to 1.5.1

**Comment by ayj** [comment]
> Pilot is not designed to be secure to the open public. There are debug interfaces, no rate limits, etc that can easily be leveraged in an attack.

istio.io also currently recommends exposing Istio services through the common gateway as an alternative to using the pod IP which is not stable. See https://preliminary.istio.io/docs/setup/install/multicluster/shared-vpn/#expose-the-istio-services-via-a-gateway. We should remove that suggestion or at least add a warning that the Istio service ports need additional protection from non-mesh sources, e.g. firewall.

**Comment by costinm** [comment]
Using the pod IPs should not even be mentioned in any doc. It's bad even for demos.

I think we should just link to the K8S Service documentation, and let people manage the Service, Gateway and VirtualService explicitly, as they do for their own applications. We just need to document the labels and ports that the Service must use to select istio:gateway deployments.

From operational perspective, users should not need to reinstall istio ( istioctl apply with cluster privs) if they just need to expose a port or create internal load balancer. Fortunately this kind of operation is well known to users, all we need is explain the ports and integration points.

**Comment by costinm** [comment]
Please check https://github.com/istio/istio.io/pull/6737 - I think we should start focusing on 'prod' deployments, where the Gateway will not be shared for mesh expansion and normal internet, and 
not require users to reapply the templates, but create an explicit K8S service or standalone Istio Gateway. Exposing Istiod is no different than exposing any other mesh service, we have explicit APIs for this without need of mixing it in the install step.

File: manifests/gateways/istio-ingress/values.yaml (hunk 1 above)
Comment (irisdingbj) [reply manifests/gateways/istio-ingress/values.yaml]: fixed. I also remove the unused virutalservice and destinationrules for pilot and citadel

**Review by howardjohn — APPROVED** [review]

**Comment by sdake** [comment]
@irisdingbj this cherrypick failed, can you please manually cherry-pick this fix.

Thanks
-steve

**Comment by sdake** [comment]
@ayj agreed. I wrote that original implementation with my friends in $priorgig$. The implementation was never fully baked sadly. I wrote some basic documentation around how to implement pilot restart protection, however, I'm not sure that was ever implemented.

I am not certain how pod IPs could be used without restart protection. As such, we should either remove this set of instructions or alter it in such a way as to correct for the RFC1912 network case. I am not sure how to properly implement the RFC1912 case without using services. I was never a fan of using the endpoint IPs...

Cheers
-steve

**Comment by irisdingbj** [comment]
> @irisdingbj this cherrypick failed, can you please manually cherry-pick this fix.
> 
> Thanks
> -steve

https://github.com/istio/istio/pull/21983

**Pull request merged**